src/sentry/snuba/snuba_query_validator.py

```diff
@@ -351,6 +359,28 @@ def _validate_performance_dataset(self, dataset):
 
         return dataset
 
+    def _validate_group_by(self, value: Sequence[str] | None) -> list[str] | None:
+        if value is None:
+            return None
+
+        if not isinstance(value, list):
+            raise serializers.ValidationError("Group by must be a list of strings")
+
+        if len(value) == 0:
+            raise serializers.ValidationError("Group by cannot be empty")
+
+        if len(value) > 100:
+            raise serializers.ValidationError("Group by must be less than 100 items")
+
+        # group by has to be unique list of strings
+        if len(value) != len(set(value)):
+            raise serializers.ValidationError("Group by must be a unique list of strings")
+
+        # TODO:
```

Comment:
group by array value can't be empty string, has to be of type string, what else?